Workbooks/SapMonitor/SapNetWeaver/SapNetWeaver.workbook

```diff
@@ -56,6 +56,36 @@
{ProviderInstance}' | where Typ_s has '{Aggregate}' | summarize totalWP = count(), freeWP = countif(Status_s == 'Wait') by TimeGenerated, Typ_s, serverTimestamp_t, AppServer = strcat(hostname_s, '_', tostring(toint(instanceNr_d))) | extend UtilizationPctWP = round(toreal(totalWP - freeWP) / toreal(totalWP) * 100, 3) | summarize AggregatedValue = max(UtilizationPctWP) by bin(TimeGenerated, 1m), AppServer, Typ_s, serverTimestamp_t | where AggregatedValue > {AlertThreshold}\",\r\n            \"aggregateName\": \"WorkType\",\r\n            \"aggregateDropDownQuery\": \"SapNetweaver_ABAPGetWPTable_CL | where Typ_s has 'DIA' or Typ_s has 'BTC' | distinct Typ_s\",\r\n            \"thresholdOperator\": \"GreaterThan\",\r\n            \"defaultThreshold\": \"70\",\r\n            \"alertUnit\": \"percent\",\r\n            \"metricMeasurement\": {\r\n                \"thresholdOperator\": \"GreaterThan\",\r\n                \"threshold\": \"2\",\r\n                \"metricTriggerType\": \"Consecutive\",\r\n                \"metricColumn\": \"AppServer\",\r\n                \"frequencyInMinutes\": \"5\",\r\n                \"timeWindowInMinutes\": \"10\"\r\n            }\r\n        }\r\n    },\r\n    {\r\n        \"name\": \"SAP Netweaver Instance Queue Wait\",\r\n        \"description\": \"Fired when SAP Netweaver Instance with DIA or BTC Queue Wait breaches specified threshold.\",\r\n        \"author\": \"Microsoft\",\r\n        \"templateId\": \"sapnetweaver-instance-queue-wait\",\r\n        \"severity\": \"2\",\r\n        \"alertTemplate\": {\r\n            \"query\": \"let workTypeCount = toscalar(SapNetweaver_GetQueueStatistic_CL | where Typ_s !contains 'ICM' | distinct Typ_s | count ); SapNetweaver_GetQueueStatistic_CL | where PROVIDER_INSTANCE_s == '{ProviderInstance}' | where Typ_s has '{Aggregate}' | summarize QueueLimit=max(Max_d)/workTypeCount by hostname_s, instanceNr_d, Typ_s, serverTimestamp_t, Now_d, TimeGenerated | extend QueueWaitPct = iff(QueueLimit != 0, (toreal(Now_d) / toreal(QueueLimit)) * 100, 0.0), AppServer = strcat(hostname_s, '_', tostring(toint(instanceNr_d))) | summarize AggregatedValue = max(QueueWaitPct) by bin(TimeGenerated, 1m), AppServer, Typ_s, serverTimestamp_t | where AggregatedValue > {AlertThreshold}\",\r\n            \"aggregateName\": \"WorkType\",\r\n            \"aggregateDropDownQuery\": \"SapNetweaver_GetQueueStatistic_CL | where Typ_s has 'DIA' or Typ_s has 'BTC' | distinct Typ_s\",\r\n            \"thresholdOperator\": \"GreaterThan\",\r\n            \"defaultThreshold\": \"5\",\r\n            \"alertUnit\": \"percent\",\r\n            \"metricMeasurement\": {\r\n                \"thresholdOperator\": \"GreaterThan\",\r\n                \"threshold\": \"2\",\r\n                \"metricTriggerType\": \"Consecutive\",\r\n                \"metricColumn\": \"AppServer\",\r\n                \"frequencyInMinutes\": \"5\",\r\n                \"timeWindowInMinutes\": \"10\"\r\n            }\r\n        }\r\n    }\r\n]",
```

Comment:
System and Instance Availability template:
In the query, looks like you are joining 2 tables based on serverTimestamp_t. Is that reliable enough to join on? Also, in the summerize section, you are binning the TimeGenerated in 2 minute buckets. Why 2 minutes?

SAP Netweaver Instance Queue Wait template:
Just a question, the default threshold is 5 percent, is this correct?

Reply:
1. Yes, to count all the app servers configured per SID and all of them have unique server timestamp 
2. Configured to capture data every 2 min in AMS, wanted tomake sure all data points are available and not to have aggregated data in specified bin
3. Yes, Its correct. Queue is shared between different work types and alerts are configured per work type. hence the configuration is on lower side.